mindsdb/integrations/handlers/rayserve_handler/rayserve_handler.py

```diff
@@ -0,0 +1,101 @@
+import base64
+import json
+import pickle
+from typing import Optional, Dict
+
+import pandas as pd
+import requests
+
+from mindsdb.integrations.libs.base import BaseMLEngine
+from mindsdb.utilities import log
+from mindsdb.interfaces.storage import db
+
+
+class RayServeHandler(BaseMLEngine):
+    name = 'rayserve'
+
+    ARG_TARGET = "target"
+    ARG_COLUMN_SEQUENCE = "column_sequence"
+    ARG_TRAIN_URL = "train_url"
+    ARG_PREDICT_URL = "predict_url"
+    ARG_PREDICTOR_ID = "predictor_id"
+    ARG_PREDICTOR_NAME = "predictor_name"
+
+    KWARGS_DF = "df"
+
+    PERSIST_ARGS_KEY_IN_JSON_STORAGE = "rayserve_args"
+
+    def create(self, target, args=None, **kwargs):
+        # prepare arguments
+        using_args = args.get("using", dict())
+        train_url = using_args.get("train_url", None)
+        predict_url = using_args.get("predict_url", None)
+        if train_url is None:
+            raise Exception("Missing required argument: using.train_url")
+        if predict_url is None:
+            raise Exception("Missing required argument: using.predict_url")
+
+        predictor_id = self.model_storage.predictor_id
+        predictor_name = db.Predictor.query.get(predictor_id).name
+        # prepare args to serialize
+        serialize_args = dict()
+        serialize_args[self.ARG_TARGET] = target
+        serialize_args[self.ARG_TRAIN_URL] = train_url
+        serialize_args[self.ARG_PREDICT_URL] = predict_url
+        serialize_args[self.ARG_PREDICTOR_ID] = predictor_id
+        serialize_args[self.ARG_PREDICTOR_NAME] = predictor_name
+        # check df
+        df: pd.DataFrame = kwargs.get(self.KWARGS_DF, None)
+        if df is None:
+            raise Exception("missing required key in args: " + self.KWARGS_DF)
+        else:
+            column_sequence = sorted(list(df.columns.values))
+            df = df[column_sequence]
+            serialize_args[self.ARG_COLUMN_SEQUENCE] = column_sequence
+
+        try:
+            resp_status_code = ""
+            log.logger.info(F"Training model by RayServe, target: {target}, "
+                            F"train_url: {train_url}, predict_url: {predict_url}")
+            encoded_df = base64.encodebytes(pickle.dumps(df))
+            resp = requests.post(train_url, json={'df': encoded_df, 'target': target,
+                                                  "predictor_id": predictor_id, "predictor_name": predictor_name})
+            resp_status_code = resp.status_code
+            assert resp.status_code == 200
+            log.logger.info(F"Training model by RayServe success.")
```

Comment:
I'm a bit unsure this is correct. The model behind the `train_url` could potentially train very slowly, so I don't think the process is ever synchronous.

In fact, I think this should be changed to follow the Lightwood handler, where we detach the learning into a separate process that will commit training stats to the DB (e.g. accuracy, training time, etc.) once the learning process _does_ finish. Would you agree?

Reply:
yes, I've just ref to the code of rayServe in Feb. 2022.
Here. I've got a problem, if we want to record the status, then we must require another url to query the status of RayServe. Would you mind if an "url.status" parameter to be added to using block?

Reply:
Not at all! Feel free to add that parameter.

Reply:
ok, I suppose I will commit my modification next week.